pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -89,7 +89,7 @@ func amendFilterChainMatchFromInboundListener(chain *listener.FilterChain, l *li
 				log.Debugf("Intercepted inbound listener %s have neither 0 or 1 prefix ranges. Actual:  %d",
 					l.Name, len(chain.FilterChainMatch.PrefixRanges))
 			}
-			if sockAddr.Address != WildcardAddress && sockAddr.Address != WildcardIPv6Address {
+			if !node.SidecarScope.HasIngressListener() && sockAddr.Address != WildcardAddress && sockAddr.Address != WildcardIPv6Address {
 				chain.FilterChainMatch.PrefixRanges = []*core.CidrRange{util.ConvertAddressToCidr(sockAddr.GetAddress())}
```

Comment:
is there any case when this condition is hit?

Reply:
Not sure if I got the question understood correctly.

The condition is hit under circumstances as https://github.com/istio/istio/blob/0bae3b969fc6b09a36826245433638651837b6b3/pilot/pkg/networking/core/v1alpha3/listener_builder.go#L77-L79 indicates and when the PrefixRanges of the fc match of that listener but has listener's address is available and not wildcard.

This can happen when bind https://github.com/istio/istio/blob/0bae3b969fc6b09a36826245433638651837b6b3/pilot/pkg/networking/core/v1alpha3/listener.go#L248-L252 uses a (global) unitcast address, applied via https://github.com/istio/istio/blob/0bae3b969fc6b09a36826245433638651837b6b3/pilot/pkg/networking/core/v1alpha3/listener_address.go#L61.

This is exactly happened in the simulation tests, where the unicast address is populated with proxy IP addresses which is faked to "1.1.1.1".

Reply:
I mean is there ever a case when `!node.SidecarScope.HasIngressListener() && sockAddr.Address != WildcardAddress && sockAddr.Address != WildcardIPv6Address` is `true`, and therefor we execute L93?

I cannot think of any case we use CIDR match on inbound listener?

Reply:
emm... Actually I can't think of any case neither. If that's the case, we may remove https://github.com/istio/istio/blob/0bae3b969fc6b09a36826245433638651837b6b3/pilot/pkg/networking/core/v1alpha3/listener_builder.go#L87-L95 entirely.

cc @lambdai

Reply:
Good point. I remember I add the dest address prefix for outbound listeners as the verification phase and an intermediate step to move to fat virtual outbound listener.
Now that we stick to the 1 thin virtual outbound listener and multiple per service outbound listeners mode,
this destination prefix match is not required

Reply:
Ok, it's inbound...

From high level, if the inbound listener is not originated from Sidecar Ingress, I don't think there is any istio config could override the sockAddr so it should be always be wildcard